sudoku.py

```diff
@@ -0,0 +1,34 @@
+def sudoku(puzzle):
+    def is_valid(num, row, col):
```

Comment:
There is usually not a need to nest functions like this, can you pull them out so they are all in the global scope.